crates/sui-core/src/traffic_controller/mod.rs

```diff
@@ -130,49 +166,113 @@ async fn handle_error_tally(
     policy: &mut TrafficControlPolicy,
-    config: &PolicyConfig,
+    policy_config: &PolicyConfig,
+    nodefw_client: &Option<NodeFWClient>,
+    tc_config: &TrafficControllerConfig,
     tally: TrafficTally,
     blocklists: Arc<Blocklists>,
 ) {
-    // TODO -- add update of spam blocklist
-    handle_tally_impl(policy, config, tally, blocklists).await
+    let resp = policy.handle_tally(tally.clone());
+    if tc_config.delegate_spam_blocking {
+        let client = nodefw_client
+            .as_ref()
+            .expect("Expected NodeFWClient for blocklist delegation");
+        delegate_policy_response(
+            resp,
+            policy_config,
+            client,
+            tally,
+            tc_config.destination_port,
+        )
+        .await;
+    } else {
+        handle_policy_response(resp, policy_config, tally, blocklists).await
+    }
 }
 
-async fn handle_tally_impl(
-    policy: &mut TrafficControlPolicy,
-    config: &PolicyConfig,
+async fn handle_policy_response(
+    PolicyResponse {
+        block_connection_ip,
+        block_proxy_ip,
+    }: PolicyResponse,
+    PolicyConfig {
+        connection_blocklist_ttl_sec,
+        proxy_blocklist_ttl_sec,
+        ..
+    }: &PolicyConfig,
     tally: TrafficTally,
     blocklists: Arc<Blocklists>,
 ) {
-    let PolicyResponse {
-        block_connection_ip,
-        block_proxy_ip,
-    } = policy.handle_tally(tally.clone());
     if block_connection_ip {
         blocklists.connection_ips.write().insert(
             tally
                 .connection_ip
                 .expect("Expected connection IP if policy is blocking it"),
-            Utc::now()
-                + chrono::Duration::seconds(
-                    config.connection_blocklist_ttl_sec.try_into().unwrap(),
-                ),
+            Utc::now() + chrono::Duration::seconds(*connection_blocklist_ttl_sec as i64),
         );
     }
     if block_proxy_ip {
         blocklists.proxy_ips.write().insert(
             tally
                 .proxy_ip
                 .expect("Expected proxy IP if policy is blocking it"),
-            Utc::now()
-                + chrono::Duration::seconds(config.proxy_blocklist_ttl_sec.try_into().unwrap()),
+            Utc::now() + chrono::Duration::seconds(*proxy_blocklist_ttl_sec as i64),
         );
     }
 }
+
+async fn delegate_policy_response(
+    PolicyResponse {
+        block_connection_ip,
+        block_proxy_ip,
+    }: PolicyResponse,
+    PolicyConfig {
+        connection_blocklist_ttl_sec,
+        proxy_blocklist_ttl_sec,
+        ..
+    }: &PolicyConfig,
+    node_fw_client: &NodeFWClient,
+    tally: TrafficTally,
+    destination_port: u16,
+) -> Result<(), reqwest::Error> {
+    let mut addresses = vec![];
+    if block_connection_ip {
+        addresses.push(BlockAddress {
+            source_address: tally.connection_ip.unwrap().to_string(),
+            destination_port,
+            ttl: *connection_blocklist_ttl_sec,
```

Comment:
we could get rid of the *connectio_blocklist_ttl_sec usage here and just access the value without the deref if we didn't borrow these structs, i think.